lib/internal/blob.js

```diff
@@ -188,49 +228,143 @@ class Blob extends JSTransferable {
 
     const span = MathMax(end - start, 0);
 
-    return new InternalBlob(
-      this[kHandle].slice(start, start + span), span, contentType);
+    return createBlob(
+      this[kHandle].slice(start, start + span),
+      span,
+      contentType);
   }
 
-  async arrayBuffer() {
+  /**
+   * @returns {Promise<ArrayBuffer>}
+   */
+  arrayBuffer() {
+    if (!isBlob(this))
+      return PromiseReject(new ERR_INVALID_THIS('Blob'));
+
+    // If there's already a promise in flight for the content,
+    // reuse it, but only once. After the cached promise resolves
+    // it will be cleared, allowing it to be garbage collected
+    // as soon as possible.
+    if (this[kArrayBufferPromise])
+      return this[kArrayBufferPromise];
+
     const job = new FixedSizeBlobCopyJob(this[kHandle]);
 
     const ret = job.run();
+
+    // If the job returns a value immediately, the ArrayBuffer
+    // was generated synchronously and should just be returned
+    // directly.
     if (ret !== undefined)
       return PromiseResolve(ret);
 
     const {
       promise,
       resolve,
-      reject
+      reject,
     } = createDeferredPromise();
+
     job.ondone = (err, ab) => {
       if (err !== undefined)
         return reject(new AbortError());
       resolve(ab);
     };
+    this[kArrayBufferPromise] =
+    SafePromisePrototypeFinally(
+      promise,
+      () => this[kArrayBufferPromise] = undefined);
 
-    return promise;
+    return this[kArrayBufferPromise];
   }
 
+  /**
+   * @returns {Promise<string>}
+   */
   async text() {
+    if (!isBlob(this))
+      throw new ERR_INVALID_THIS('Blob');
+
     const dec = new TextDecoder();
     return dec.decode(await this.arrayBuffer());
   }
+
+  /**
+   * @returns {ReadableStream}
+   */
+  stream() {
+    if (!isBlob(this))
+      throw new ERR_INVALID_THIS('Blob');
+
+    const self = this;
+    return new lazyReadableStream({
+      async start(controller) {
+        const ab = await self.arrayBuffer();
+        controller.enqueue(new Uint8Array(ab));
```

Comment:
... not so friendly if you need to handle large data? 😕

Reply:
Yes, I know. We can revisit this. But if you check, you'll see that the behavior here matches what is implemented in browsers.

Reply:
browsers don't enqueue one large chunks - well, it can depending on how small it is or how the blob was constructed

<img width="500" src="https://user-images.githubusercontent.com/1148376/128872123-4671c346-5e1f-4705-a436-8e09779261b7.png">

Reply:
Ah, right, forgot about that. Internally it's one big ArrayBuffer but I believe it slices out the reads on a 65k max boundary in Chrome. I'll do the same.

Reply:
if it would enqueue one single large uint8array then there would be no point in having `.stream()` 😉

Reply:
a lot of features are implemented to allow optimization and do not ship optimized in the first iteration. I think we can apply that approach to various things here to land the API.